TheMET/TheMET/Cache/ImageCache.swift

```diff
@@ -39,7 +39,9 @@ class ImageCache {
                     return
                 }
                 self.loadedImagesCollection[imageURL] = image
-                completion(image)
+                DispatchQueue.main.async {
+                    completion(image)
```

Comment:
All the completions executed which are currently executed on `DispatchQueue.global` in this thread must be executed back on `main`